### PR TITLE
[MAPLE-743] Add non blocking to hand_pose and change default traj timing

### PR DESCRIPTION
**Pull request by llee-bdai**
Add in non blocking option to return before arm trajectory completes.

Also changes to the default trajectory timing to 0.0 (as fast as possible) unless specifically specified to be some duration.

### CODE REVIEW

File: spot_wrapper/spot_arm.py [hunk 1]
```diff
@@ -447,9 +447,10 @@ def hand_pose(
         qy: float,
         qz: float,
         qw: float,
-        duration: float = 0.125,
+        duration: float = 0.0,
```
Comment (scastro-bdai) [inline spot_wrapper/spot_arm.py:450]: Will this cause an incompatibility with other users who may be relying on the nonzero default arg?

File: spot_wrapper/spot_arm.py [hunk 2]
```diff
@@ -483,11 +484,14 @@ def hand_pose(
             # Rotation as a quaternion.
             rotation = geometry_pb2.Quaternion(w=qw, x=qx, y=qy, z=qz)
 
-            traj_duration = seconds_to_duration(duration)
-
             # Build the SE(3) pose of the desired hand position in the moving body frame.
             hand_pose = geometry_pb2.SE3Pose(position=position, rotation=rotation)
-            hand_pose_traj_point = trajectory_pb2.SE3TrajectoryPoint(pose=hand_pose, time_since_reference=traj_duration)
+            hand_pose_traj_point = trajectory_pb2.SE3TrajectoryPoint(pose=hand_pose)
+            if duration != 0.0:
```
Comment (scastro-bdai) [inline spot_wrapper/spot_arm.py:490]: Generally skeptical of floating point equality checks. If it were up to me, I'd maybe make `duration` be a `double | None`, so you can specifically set it to `None` or something.

File: spot_wrapper/spot_arm.py (hunk 1 above)
Comment (llee-bdai) [reply spot_wrapper/spot_arm.py]: Seeing as I am the only user right now.... probably not?

File: spot_wrapper/spot_arm.py (hunk 2 above)
Comment (llee-bdai) [reply spot_wrapper/spot_arm.py]: Updated to be optionally None. Thanks for catching that, I wasn't really going through everything with a fine toothcomb

**Review by scastro-bdai — APPROVED** [review]

**Comment by llee-bdai** [comment]
### Merge activity

* **Oct 17, 10:13 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/bdaiinstitute/spot_wrapper/146).

**Pull request merged**